File: fetch_genres.js
const http = require('http');

async function fetchJson(url) {
  return new Promise((resolve, reject) => {
    http.get(url, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        try { resolve(JSON.parse(data)); }
        catch (e) { reject(e); }
      });
    }).on('error', reject);
  });
}

async function run() {
  const genresRes = await fetchJson('http://localhost:5000/api/tmdb/genres');
  if (!genresRes.genres) {
    console.error('No genres found');
    return;
  }
  
  const mapping = {};
  for (const g of genresRes.genres) {
    try {
      const page = await fetchJson(`http://localhost:5000/api/tmdb/discover?with_genres=${g.id}&sort_by=popularity.desc`);
      if (page && page.results && page.results.length > 0) {
        // Find first movie with backdrop
        const movie = page.results.find(m => m.backdrop_path) || page.results[0];
        if (movie && movie.backdrop_path) {
          mapping[g.name] = `https://image.tmdb.org/t/p/w780${movie.backdrop_path}`;
        }
      }
    } catch (e) {
      console.error('Failed for genre', g.name, e.message);
    }
  }
  
  console.log(JSON.stringify(mapping, null, 4));
}
run();

File: server/fetch_genres.py
import urllib.request
import json
import time

import os

API_KEY = os.environ.get('TMDB_API_KEY')
base = 'https://api.themoviedb.org/3'

def get_json(url):
    for i in range(3):
        try:
            req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
            with urllib.request.urlopen(req) as res:
                return json.loads(res.read().decode('utf-8'))
        except Exception as e:
            time.sleep(1)
    return {}

data = get_json(f'{base}/genre/movie/list?api_key={API_KEY}')
mapping = {}
for g in data.get('genres', []):
    page = get_json(f'{base}/discover/movie?api_key={API_KEY}&with_genres={g["id"]}&sort_by=popularity.desc')
    for m in page.get('results', []):
        if m.get('backdrop_path'):
            mapping[g['name']] = f'https://image.tmdb.org/t/p/w780{m["backdrop_path"]}'
            break

print(json.dumps(mapping, indent=4))
